### PR TITLE
97 issues with client facing header files 2

### LINKED ISSUE #97: Issues with Client facing header files
The following issues came up when trying to hack a pythong chronolog client:
1. We currently build the binary libchronolog_client.so library that is placed under ../build/Client by default but we do not create a client facing include directory for the client apps to use ( aka ../build/Client/include).  All the header files that are needed to build the client application should be copied to this directory as part of building  chronolog client library; at he moment these files are chronolog_client.h, ConfigurationManager.h, and errcode.h.
2. errcode.h  is such a generic filename that is clashes with Python errcode.h, this file needs to be renamed and the errcodes need to be redefined as chronolog namespace enum object instead of DEFINE macros
3. all hte tests that we have under test/integration , and any future client application examples we build should be built using hte client facing INCLUDE dir instead of referencing the header files from  ChronoLogClient/include/ , ChronoLog/ChronoAPI/include/, ChronoLog/chrono_common, etc, so that if we introduce any new client facing headers in the future and do not propagate them into the client facing include directory as part of the library build process, our own tests would brake and the issue would be exposed.

### PR DESCRIPTION
**Pull request by iameneko**
Issue
Closes #97

### Changes Made:

**1. Client Facing Include Directory:**
- Added the creation of a client-facing include directory (../build/Client/include) as part of the build process for libchronolog_client.so.
- Copied necessary header files (chronolog_client.h, ConfigurationManager.h, and errcode.h) to the client-facing include directory.

**2. Renaming and Refactoring:**
- Renamed errcode.h to chronlog_errcode.h (to avoid conflicts).
- Refactored errcode.h to use the chronolog namespace for enum objects instead of DEFINE macros.

**3. Test and Example Updates:**
- Updated all tests under test/integration to use the client-facing include directory for referencing header files.
- Ensured that client application examples reference the header files from the client-facing include directory.

### Notes: 
All the changes made have been tested by running the following tests: chronovisor_server, chrono_keeper and client_lib_multi_storytellers

### CODE REVIEW

File: ChronoAPI/ChronoLog/include/chronolog_errcode.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// Created by kfeng on 12/19/22.
+//
+
+#ifndef CHRONOLOG_CHRONOLOG_ERRCODE_H
+#define CHRONOLOG_CHRONOLOG_ERRCODE_H
+namespace chronolog
+{
+enum ErrorCode
+{
+    CL_SUCCESS = 0,                      // Success
```
Comment (fkengun) [inline ChronoAPI/ChronoLog/include/chronolog_errcode.h:11]: Please keep a similar alignment as the original errocode.h. I think that is more readable. If any line is longer than 120, pls adjust the spaces between error names and =.

File: test/integration/Client/client_lib_multi_argobots_test.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ void thread_function(void*tt)
     chronicle_attrs.emplace("TieringPolicy", "Hot");
     ret = client->CreateChronicle(chronicle_name, chronicle_attrs, flags);
     std::cout << "tid=" << t->tid << " Createchronicle {" << chronicle_name << "} ret: " << ret << std::endl;
-    assert(ret == CL_SUCCESS || ret == CL_ERR_CHRONICLE_EXISTS || ret == CL_ERR_NO_KEEPERS);
+    assert(ret == chronolog::CL_SUCCESS || ret == chronolog::CL_ERR_CHRONICLE_EXISTS || ret == chronolog::CL_ERR_NO_KEEPERS);
```
Comment (fkengun) [inline test/integration/Client/client_lib_multi_argobots_test.cpp:43]: This line is longer than 120. Please reformat it. Please do the same for all other test case files.

File: test/integration/Client/client_lib_multi_pthread_test.cpp [hunk 3]
```diff
@@ -42,25 +42,25 @@ void thread_body(struct thread_arg*t)
...
     ret = client->ReleaseStory(chronicle_name, story_name);//, flags);
     std::cout << "tid=" << t->tid << " ReleaseStory {" << chronicle_name << ":" << story_name << "} ret: " << ret
               << std::endl;
-    assert(ret == CL_SUCCESS || ret == CL_ERR_NO_CONNECTION);
+    assert(ret == chronolog::CL_SUCCESS || ret == CL_ERR_NO_CONNECTION);
     ret = client->DestroyStory(chronicle_name, story_name);//, flags);
     std::cout << "tid=" << t->tid << " DestroyStory {" << chronicle_name << ":" << story_name << "} ret: " << ret
               << std::endl;
-    assert(ret == CL_SUCCESS || ret == CL_ERR_NOT_EXIST || ret == CL_ERR_ACQUIRED || ret == CL_ERR_NO_CONNECTION);
+    assert(ret == chronolog::CL_SUCCESS || ret == CL_ERR_NOT_EXIST || ret == CL_ERR_ACQUIRED || ret == CL_ERR_NO_CONNECTION);
```
Comment (fkengun) [inline test/integration/Client/client_lib_multi_pthread_test.cpp:59]: Long line.

File: test/integration/Client/client_lib_multi_storytellers.cpp [hunk 4]
```diff
@@ -48,14 +48,14 @@ void thread_body(struct thread_arg*t)
...
         std::cout << "tid=" << t->tid << " ReleaseStory {" << chronicle_name << ":" << story_name << "} ret: " << ret
                   << std::endl;
-        assert(ret == CL_SUCCESS || ret == CL_ERR_NO_CONNECTION);
+        assert(ret == chronolog::CL_SUCCESS || ret == chronolog::CL_ERR_NO_CONNECTION);
     }
     ret = client->DestroyStory(chronicle_name, story_name);//, flags);
     std::cout << "tid=" << t->tid << " DestroyStory {" << chronicle_name << ":" << story_name << "} ret: " << ret
               << std::endl;
-    assert(ret == CL_SUCCESS || ret == CL_ERR_NOT_EXIST || ret == CL_ERR_ACQUIRED || ret == CL_ERR_NO_CONNECTION);
+    assert(ret == chronolog::CL_SUCCESS || ret == chronolog::CL_ERR_NOT_EXIST || ret == chronolog::CL_ERR_ACQUIRED || ret == chronolog::CL_ERR_NO_CONNECTION);
```
Comment (fkengun) [inline test/integration/Client/client_lib_multi_storytellers.cpp:56]: Long line.

File: test/integration/Client/client_lib_multi_pthread_test.cpp (hunk 3 above)
Comment (fkengun) [inline test/integration/Client/client_lib_multi_pthread_test.cpp]: CL_ERR_NOT_EXIST and CL_ERR_ACQUIRED are not updated to chronolog namespace.

File: test/integration/Client/client_lib_multi_pthread_test.cpp [hunk 5]
```diff
@@ -42,25 +42,25 @@ void thread_body(struct thread_arg*t)
...
-    assert(ret == CL_ERR_ACQUIRED || ret == CL_SUCCESS || ret == CL_ERR_NOT_EXIST);
+    assert(ret == chronolog::CL_ERR_ACQUIRED || ret == chronolog::CL_SUCCESS || ret == chronolog::CL_ERR_NOT_EXIST);
     ret = client->Disconnect(); //t->client_id, flags);
-    assert(ret == CL_ERR_ACQUIRED || ret == CL_SUCCESS);
+    assert(ret == chronolog::CL_ERR_ACQUIRED || ret == chronolog::CL_SUCCESS);
     ret = client->ReleaseStory(chronicle_name, story_name);//, flags);
     std::cout << "tid=" << t->tid << " ReleaseStory {" << chronicle_name << ":" << story_name << "} ret: " << ret
               << std::endl;
-    assert(ret == CL_SUCCESS || ret == CL_ERR_NO_CONNECTION);
+    assert(ret == chronolog::CL_SUCCESS || ret == CL_ERR_NO_CONNECTION);
```
Comment (fkengun) [inline test/integration/Client/client_lib_multi_pthread_test.cpp]: CL_ERR_NO_CONNECTION is not updated to chronolog namespace.

**Review by fkengun — CHANGES_REQUESTED** [review]
pls fix error code in test cases.

File: ChronoAPI/ChronoLog/include/chronolog_errcode.h (hunk 1 above)
Comment (iameneko) [reply ChronoAPI/ChronoLog/include/chronolog_errcode.h:11]: I do not understand this properly. Do you mean having all numbers in the same column?
The space between error names and = is the one established in the code style. I mean, the file is based on code style. Do I have something wrong on my format?

Comment (fkengun) [reply ChronoAPI/ChronoLog/include/chronolog_errcode.h:11]: I thought the style allows a aligned number column. but it does not. then pls keep the current version.

**Review by fkengun — APPROVED** [review]

**Review by kmk142789 — APPROVED** [review]

**Pull request merged**